63/Gealber/generate_blog_images.py

```diff
@@ -0,0 +1,128 @@
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.keys import Keys
+import random
+import time
+import zipfile
+import os
+import re
+
+
+class Extractor:
+    def __init__(self, headers):
+        self.headers = headers
+
+    @staticmethod
+    def parser(page, query):
+        soup = BeautifulSoup(page, 'lxml')
+        items = soup.select(query)
+        return items
+
+    def extract_challenge(self):
+        url = 'https://codechalleng.es/challenges/'
+        request = Request(url, headers=self.headers)
+        base_url = 'https://codechalleng.es'
+        challenges = {}
+        try:
+            with urlopen(request) as response:
+                html = str(response.read(), 'utf-8')
+            items = self.parser(html, "td.challengeTitle > a")
+            for item in items:
+                link = "".join([base_url, item.get('href')])
+                challenges[item.get_text()] = link
+        except URLError as error:
+            print(error.reason)
+        return challenges
+
+    def extract_articles(self):
+        url = 'https://pybit.es/pages/articles.html'
+        request = Request(url, headers=self.headers)
+        articles = {}
+        try:
+            with urlopen(request) as response:
+                html = str(response.read(), 'utf-8')
+            items = self.parser(html, "ul.articleList > li > a")
+            for item in items:
+                link = item.get('href')
+                articles[item.get_text()] = link
+        except URLError as error:
+            print(error.reason)
+
+        return articles
+
+
+class ImageGenerator:
+    def __init__(self, driver_executable):
+        self.driver_executable = driver_executable
+
+    @staticmethod
+    def fill_form(driver, title: str, url: str):
+        try:
+            title_element = driver.find_element_by_id("title")
+            title_element.send_keys(Keys.SHIFT, Keys.HOME, Keys.BACKSPACE)
+            # title_element.send_keys(Keys.BACKSPACE * 15)
+            title_element.send_keys(title)
+            topoff_select = Select(driver.find_element_by_id("topoffset"))
+            time.sleep(1)
+            topoff_select.select_by_value("10px")
+            theme_select = Select(driver.find_element_by_id("collection"))
+            time.sleep(1)
+            theme_select.select_by_value("bamboo")
+            image_select = driver.find_element_by_id("bg1_url")
+            image_select.click()
+            time.sleep(5)
+            image_select.send_keys(Keys.DOWN * random.randint(1, 25))
+            image_select.send_keys(Keys.ENTER)
+            overlay_url = driver.find_element_by_id("overlay_url")
+            overlay_url.send_keys(Keys.SHIFT, Keys.HOME, Keys.BACKSPACE)
+            overlay_url.send_keys(url)
+            save = driver.find_element_by_id("btnSave")
+            save.click()
+            time.sleep(5)
+        except Exception as e:
```

Comment:
what exception can this raise, make try/except as narrow as possible around the "risky" code